api/src/OpenTelemetry/Trace/Core.hs

```diff
@@ -285,10 +286,13 @@ createSpanWithoutCallStack t ctxt n args@SpanArguments {..} = liftIO $ do
                     , spanEnd = Nothing
                     , spanTracer = t
                     }
+
+            when (A.attributesDropped (spanAttributes is) > 0) $ void logDroppedAttributes
+
             s <- newIORef is
             eResult <- try $ mapM_ (\processor -> processorOnStart processor s ctxt) $ tracerProviderProcessors $ tracerProvider t
             case eResult of
-              Left err -> print (err :: SomeException)
+              Left err -> void $ emitOTelLogRecord H.empty Err $ T.pack $ show (err :: SomeException)
```

Comment:
With the `print` statement, this would still be in effect "logged" even if logging is no-op. The code should probably be `grep`ed for print statements that print errors.